lib/ansible/modules/database/mongodb/mongodb_replicaset.py

```diff
@@ -0,0 +1,412 @@
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: mongodb_replicaset
+short_description: Initialises a MongoDB replicaset before authentication has been turned on and then validates the configuration when it has been turned on.
+description:
+    - Initialises a MongoDB replicaset before authentication has been turned on and then validates the configuration when it has been turned on.
+    - Validation takes the form of replicaset name validation only at present.
+    - This may change in the future to involve adding  and removing members.
+author: Rhys Campbell (@rhysmeister)
+version_added: "2.8"
+options:
+    login_user:
+        description:
+            - The username used to authenticate with
+        required: false
+    login_password:
+        description:
+            - The password used to authenticate with
+        required: false
+    login_database:
+        description:
+            - The database where login credentials are stored
+        required: false
+        default: "admin"
+    login_host:
+        description:
+            - The MongoDB hostname to login to
+        required: false
+        default: "localhost"
+    login_port:
+        description:
+            - The MongoDB port to login to
+        required: false
+        default: 27017
+    replica_set:
+        description:
+            - Replicaset name
+        required: false
+        default: rs0
+    members:
+        description:
+            - An comma sepaeated list consisting of the replicaset members.
+            - Supply as a simple csv string
+                mongodb1:27017,mongodb2:27017,mongodb3:27017
+    validate:
+        description:
+            - Performs some basic validation on the provided replicaset config
+        required: false
+        default: True
+        type: bool
+    ssl:
+        description:
+            - Whether to use an SSL connection when connecting to the database
+        default: False
+        type: bool
+    ssl_cert_reqs:
+        description:
+            - Specifies whether a certificate is required from the other side of the connection, and whether it will be validated if provided.
+        required: false
+        default: "CERT_REQUIRED"
+        choices: ["CERT_REQUIRED", "CERT_OPTIONAL", "CERT_NONE"]
+    arbiter_at_index:
+        description: Identifies the position of the member in the array that is an arbiter.
+        required: false
+    chainingAllowed:
+        description: >
+            When settings.chainingAllowed is true, the replica set allows secondary members to replicate from other
+            secondary members. When settings.chainingAllowed is false, secondaries can replicate only from the primary.
```

Comment:
When referring to settings (for example, `settings.chainingAllowed` here and in the line above), you can make them stand out with formatting - see  https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html#linking-within-module-documentation for details.